packages/features/bookings/lib/handleNewBooking/checkBookingAndDurationLimits.ts

```diff
@@ -40,4 +44,28 @@ export const checkBookingAndDurationLimits = async ({
       );
     }
   }
+
+  if (eventType.userId && !eventType.schedulingType) {
+    const eventTypeUser = await prisma.user.findUnique({
+      where: {
+        id: eventType.userId,
+      },
+      select: {
+        id: true,
+        email: true,
+        bookingLimits: true,
+      },
+    });
+    if (eventTypeUser?.bookingLimits && Object.keys(eventTypeUser.bookingLimits).length > 0) {
+      await checkBookingLimits(
+        eventTypeUser.bookingLimits as IntervalLimit,
+        startAsDate,
+        eventType.id,
+        reqBodyRescheduleUid,
+        eventType.schedule?.timeZone,
+        { id: eventTypeUser.id, email: eventTypeUser.email },
```

Comment:
here we correctly pass the user, that's why we get this error on yearly limit (slots are shown that shouldn't be available)
![Screenshot 2024-10-15 at 10 13 42 AM](https://github.com/user-attachments/assets/2b21003d-c718-4381-92dc-5142bbdeabfd)